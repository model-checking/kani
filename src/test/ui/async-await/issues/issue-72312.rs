// edition:2018
fn require_static<T: 'static>(val: T) -> T {
    //~^ NOTE 'static` lifetime requirement introduced by this bound
    val
}

struct Problem;

impl Problem {
    pub async fn start(&self) { //~ ERROR E0759
        //~^ NOTE this data with an anonymous lifetime `'_`
        //~| NOTE in this expansion of desugaring of `async` block or function
        require_static(async move { //~ NOTE ...and is required to live as long as `'static` here
            &self; //~ NOTE ...is used here...
        });
    }
}

fn main() {}
