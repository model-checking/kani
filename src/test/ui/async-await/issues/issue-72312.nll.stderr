error[E0521]: borrowed data escapes outside of associated function
  --> $DIR/issue-72312.rs:13:24
   |
LL |       pub async fn start(&self) {
   |                          -----
   |                          |
   |                          `self` is a reference that is only valid in the associated function body
   |                          let's call the lifetime of this reference `'1`
...
LL |           require_static(async move {
   |  ________________________^
LL | |             &self;
LL | |         });
   | |         ^
   | |         |
   | |_________`self` escapes the associated function body here
   |           argument requires that `'1` must outlive `'static`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0521`.
